copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopyBookSource.java

```diff
@@ -0,0 +1,291 @@
+      "        {\n" +
+      "            \"name\": \"age\",\n" +
+      "            \"type\": \"int\"\n" +
+      "        }" +
+      "    ]\n" +
+      "}")
+    private String schema;
+
+    @Name("fileStructure")
+    @Nullable
+    @Description("Copybook file structure")
+    private String fileStructure;
+
+    @Name("copybookContents")
+    @Description("Contents of the cobol Copybook file which will contain " +
+      "the data structure. Eg: \n" +
+      "000100*                                                                         \n" +
+      "000200*   DTAR020 IS THE OUTPUT FROM DTAB020 FROM THE IML                       \n" +
+      "000300*   CENTRAL REPORTING SYSTEM                                              \n" +
+      "000400*                                                                         \n" +
+      "000500*   CREATED BY BRUCE ARTHUR  19/12/90                                     \n" +
+      "000600*                                                                         \n" +
+      "000700*   RECORD LENGTH IS 27.                                                  \n" +
+      "000800*                                                                         \n" +
+      "000900        03  DTAR020-KCODE-STORE-KEY.                                      \n" +
+      "001000            05 DTAR020-KEYCODE-NO      PIC X(08).                         \n" +
+      "001100            05 DTAR020-STORE-NO        PIC S9(03)   COMP-3.               \n" +
+      "001200        03  DTAR020-DATE               PIC S9(07)   COMP-3. ")
+    private String copybookContents;
+
+    public CopybookSourceConfig(String referenceName, String copybookContents, String binaryFilePath, String schema,
+                                String fileStructure) {
+      super(referenceName);
+      this.copybookContents = copybookContents;
+      this.binaryFilePath = binaryFilePath;
+      this.schema = schema;
+      this.fileStructure = fileStructure;
+    }
+
+    public CopybookSourceConfig() {
+      super(String.format("CopybookReader"));
+      fileStructure = Integer.toString(DEFAULT_FILE_STRUCTURE);
+    }
+
+    @VisibleForTesting
+    public final String getFileStructure() {
+      return fileStructure;
+    }
+
+    /**
+     * Validate the configuration parameters required
+     */
+    private void validate() {
+      if (schema != null) {
+        parseSchema();
+      }
+      //check the file extension
+      if (binaryFilePath.length() < 4) {
+        throw new IllegalArgumentException("Invalid binary file path: " + binaryFilePath);
+      } else {
+        if (!(binaryFilePath.substring(binaryFilePath.length() - 4).equals(".bin"))) {
+          throw new IllegalArgumentException("Invalid binary file path: " + binaryFilePath);
+        }
+      }
+      if (!fileStructure.isEmpty()) {
+        try {
+          Preconditions.checkArgument(
+            SUPPORTED_FILE_STRUCTURES.contains(Integer.parseInt(fileStructure)),
```

Comment:
For the initial implementation only fixed-length flat files are used.
Kept this considering future implementations where there could be multiple file types.